nix/launch-migration-test.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# Run the migration-test executable and launch a self-node.
+# You should have all necessary programs on the PATH.
+# Set the "stateDir" environment variable to a suitable temporary location.
+
+set -euo pipefail
+
+if [ -z "${1:-}" ]; then
+  echo "usage: $0 (run | step1 | step2)"
+  exit 1
+fi
+
+if [ -z "${stateDir:-}" ]; then
+  echo "$0: set the stateDir environment variable before running this script."
+  exit 1
+fi
+
+: "${src:=$(cd `dirname $0`/..; pwd)}"
```

Comment:
Slightly magic at first glance. Had to look up the semantic of the colon `:` which actually still feels like a magic trick (though I guess most bash scripts are in a way a condensed collection of magic tricks :upside_down_face: )

Why not simply ```src=$(cd `dirname $0`/..; pwd)``` ? Do we expect `src` to be possibly defined globally ?

Reply:
The `src` environment variable that can be set externally so that the script can be used both inside and outside of a nix build.

I thought the `:` for variable default values was standard well-known bash awfulness.

Reply:
I was not referring to the `:` in `src:=`, but to the leading one. First time I see this of my life to be honest :sweat_smile: ... and I've written quite some bash already.